llvm/lib/ObjCopy/ELF/ELFObject.cpp

```diff
@@ -2206,8 +2206,16 @@ Error Object::removeSections(
   // Transfer removed sections into the Object RemovedSections container for use
   // later.
   std::move(Iter, Sections.end(), std::back_inserter(RemovedSections));
-  // Now finally get rid of them all together.
+  // Now get rid of them all together.
   Sections.erase(Iter, std::end(Sections));
+
+  // Finally iterate over all sections and erase empty SHT_GROUP sections.
+  for (auto Iter = Sections.begin(); Iter != Sections.end(); ++Iter) {
```

Comment:
At least for the for loop, I am not convinced that braces are needed. Furthermore:
https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop

Reply:
Or:
```
llvm::remove_if(Sections, Lambda_Predicate);
```

Reply:
Removed the braces. The loop cannot be rewritten to `auto I = Sections.begin(), E = Sections.End(); I != E; ++I` because `Sections` may change size every time

Reply:
`llvm::remove_if` as i can see isn't very widely used so i think we can leave `for` here